skops/card/tests/test_card.py

```diff
@@ -169,6 +172,126 @@ def test_add_plot(destination_path, model_card):
     assert "![fig1](fig1.png)" in model_card
 
 
+def test_permutation_importances(
+    iris_estimator, iris_data, model_card, destination_path
+):
+    X, y = iris_data
+    result = permutation_importance(
+        iris_estimator, X, y, n_repeats=10, random_state=42, n_jobs=2
+    )
+
+    model_card.add_permutation_importances(
+        result,
+        X.columns,
+        Path(destination_path) / "importance.png",
+        "Permutation Importance",
+    )
+    temp_path = Path(destination_path) / "importance.png"
+    assert f"![Permutation Importance]({temp_path}" in model_card.render()
+
+
+def test_matplotlib_dependency_error(
+    iris_estimator, iris_data, model_card, destination_path
+):
+    orig_import = __import__
+
+    def mock_import(name, *args):
+        if name == "matplotlib.pyplot":
+            pass
+        else:
+            return orig_import(name, *args)
+
+    X, y = iris_data
+    result = permutation_importance(
+        iris_estimator, X, y, n_repeats=10, random_state=42, n_jobs=2
+    )
+
+    with mock.patch("builtins.__import__", side_effect=mock_import):
+        with pytest.raises(ImportError, match="cannot import name*"):
```

Comment:
Hmm, I'm not sure if the test does what it's supposed to do. First, I don't understand why we need two `pytest.raises`. Second, I would have thought that we need to raise `ModuleNotFoundError` in the mock for matplotlib. Third the `except` in the actual function body appears to not be reached at all according to coverage:

> Added line #L414 was not covered by tests

Note that one problem with implementing this test properly could be that `patch` is applied too late. This can happen if matplotlib was already imported at the time that the patch is applied, since Python will not attempt to be imported again, if I'm not mistaken. One way to find this out is if the test behaves the same when run in isolation (i.e. `pytest ... -k test_matplotlib_dependency_error` vs running all tests).

Reply:
we probably should just create a utility function like `import_or_raise`, and test that once, and in other methods just use that one.